WordPress/Classes/ViewRelated/Site Creation/Shared/ErrorStates/ErrorStateViewConfiguration.swift

```diff
@@ -0,0 +1,84 @@
+
+import Foundation
+
+// MARK: ErrorViewType
+
+/// Describes the possible states of an error view presented during the Site Creation flow.
+///
+/// - general:              Describes an unforeseen problem occurred.
+/// - siteLoading:          Describes a error that specifically occurred while attempting to create a site.
+/// - networkUnreachable:   Describes a case where network connectivity was explicitly not available (see also : `NetworkAware`)
+///
+enum ErrorStateViewType {
```

Comment:
I am not a big fan of types that exhibit different behaviours according to a type parameter (which I associate with one of the refactorings in Martin Fowler's book,[ Replace conditional with polymorphism](https://robots.thoughtbot.com/refactoring-replace-conditional-with-polymorphism)). Would it make sense to delayer ErrorStateView as a protocol, with different implementations for each actual type of error?

Reply:
If you find that you'd like to extract a protocol as you integrate this with the other services, I think that's an ideal future enhancement.